src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -926,12 +926,22 @@ public static bool TryCompleteSendTo(SafeSocketHandle socket, ReadOnlySpan<byte>
                         Debug.Assert(buffers == null);
                         sent = SysWrite(socket, buffer, ref offset, ref count, out errno);
                     }
+                    else if (buffers is not null)
+                    {
+                        sent = SysSend(socket, flags, buffers, ref bufferIndex, ref offset, socketAddress, socketAddressLen, out errno);
+                    }
+                    else if (buffer.Length == 0)
+                    {
+                        sent = 0;
+                        errno = Interop.Error.SUCCESS;
+                    }
```

Comment:
Are there any SocketFlags that this could potentially cause problems for?  Wondering if instead of this, we should ensure we still make the syscall so that the OS can do whatever it needs to do, and just substitute in Array.Empty for buffer.

Reply:
It's possible. Your suggestion would be safer, for sure.

On the other hand, if someone does a 0-byte send today, this change will save a syscall.

On the other other hand, don't do 0-byte sends.

Reply:
> substitute in Array.Empty for buffer

If we did this we'd still need to pin Array.Empty, right? Is that bad?

Reply:
> If we did this we'd still need to pin Array.Empty, right? Is that bad?

For the duration of the synchronous syscall.  That's fine.

That said, it doesn't need to be Array.Empty.  It could just be a stack address:
```C#
byte b = 0;
buffer = MemoryMarshal.CreateReadOnlySpan(ref b, 0);
SysSend(..., buffer, ...);
```

Reply:
I pushed a new fix. PTAL.